src/NumericalAlgorithms/DiscontinuousGalerkin/Actions/ComputeVolume.hpp

```diff
@@ -0,0 +1,81 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <tuple>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Requires.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+// IWYU pragma: no_forward_declare db::DataBox
+namespace tuples {
+template <typename...>
+class TaggedTuple;  // IWYU pragma: keep
+}  // namespace tuples
+
+namespace Parallel {
+template <typename Metavariables>
+class ConstGlobalCache;
+}  // namespace Parallel
+/// \endcond
+
+namespace dg {
+namespace Actions {
+/*!
+ * \ingroup ActionsGroup
+ * \ingroup DiscontinuousGalerkinGroup
+ * \brief Compute the volume contribution to the temporal step of the variables
+ *
+ * The `system::volume` operator is invoked with its `argument_tags`. The result
+ * of this computation is stored in `db::add_tag_prefix<step_prefix,
+ * variables_tag>`, where `step_prefix` is retrieved from the
+ * `Metavariables::temporal_id`. For instance, in a time evolution this prefix
+ * would represent a time derivative `Tags::dt`.
+ *
+ * Uses:
+ * - Metavariables:
+ *   - `temporal_id::step_prefix`
+ * - System:
+ *   - `variables_tag`
+ *   - `volume`
+ * - DataBox:
+ *   - db::add_tag_prefix<step_prefix, variables_tag>
+ *   - All elements in `volume::argument_tags`
+ *
+ * DataBox changes:
+ * - Modifies:
+ *   - `db::add_tag_prefix<step_prefix, variables_tag>`
+ */
+template <typename Metavariables>
+struct ComputeVolume {
```

Comment:
If we keep this in the `dg` namespace then this means the volume contribution to the dg discretisation. Only if we try to be too general with this does it become ambiguous I think. But if we really want to generalize we may as well go all the way and have an `Action::MutateApply` that works with any struct that has an appropriate interface for `db::mutate_apply`.

Reply:
A general action might be the right way to go here, yeah.

Reply:
Yes, @nilsdeppe and I have also discussed introducing an `Action::MutateApply` as this pattern occurs frequently.  If it allows us to avoid an ambiguous name in this case, maybe it is worth introducing now.